python/pyarrow/feather.py

```diff
@@ -111,9 +120,9 @@ def write_feather(df, dest):
         raise
 
 
-def read_feather(source, columns=None, nthreads=1):
+def read_feather(source, columns=None, nthreads=1, as_table=False):
```

Comment:
I would also add a `read_table` method here as top-level function and keep the `read_feather` as-is. This will make it in line from an API perspective with the `pyarrow.parquet` module.